include/nix/Block.hpp

```diff
@@ -446,12 +446,21 @@ class NIXAPI Block : public base::EntityWithMetadata<base::IBlock> {
                               const std::string &type,
                               const std::vector<Column> &cols,
                               const Compression &compression=Compression::Auto) {
+        std::vector<std::string> names;
+
         for (const Column &c : cols) {
+            names.push_back(c.name);
             if (!Variant::supports_type(c.dtype)) {
                 std::string msg = "Incompatible DataType for column ";
                 throw std::invalid_argument(msg + c.name);
             }
         }
+        std::sort(names.begin(), names.end());
+        names.erase(std::unique(names.begin(), names.end()), names.end());
+        if (names.size() != cols.size()) {
+            throw ConsistencyError("Block::createDataFrame: Column names must be unique!");
+        }
+
```

Comment:
I am ok with this but wouldn't using a [`set`](https://en.cppreference.com/w/cpp/container/set) be more convenient?

Reply:
Yes, of course, thanks for pointing me to it!